src/hermes/model/merge.py

```diff
@@ -131,7 +134,14 @@ def __call__(self, target, path, value, **kwargs):
                 match target[key]:
                     case dict() as item: item.update(value)
                     case list() as item: item[:] = value
-                    case _: target[key] = value
+                    case _:
+                        if key in target and target[key] != value:
```

Comment:
`if not key in target`, the access in line 134 should already throw, so, this check is superfluous.